### PR TITLE
Change colors

### LINKED ISSUE #35: Change colors
- Change colors to white and black.
- White always play first.
- Use `W` and `B` in the board instead of `red` and `blue`

### PR DESCRIPTION
**Pull request by apojomovsky**
Fixes #35 
- Replaced blue with white, red with black
- Replaced strings (like 'blue' and 'red') by letters B and W on matrix
- Adapted most of the gameboard tests to use the board_builder (cannot get rid of the from_matrix method yet)

### CODE REVIEW

File: game/gameboard.py [hunk 1]
```diff
@@ -47,10 +47,10 @@ def __str__(self):
             for entry in row:
                 if entry is None:
                     representation = representation + '. '
-                elif entry == 'red':
-                    representation = representation + 'R '
-                elif entry == 'blue':
+                elif entry == 'B':
```
Comment (basicNew) [inline game/gameboard.py]: why not ?
```else: 
 representation = representation + entry + ' '
```

File: tests/test_match.py [hunk 2]
```diff
@@ -9,66 +9,66 @@
 
 class TestMatch(unittest.TestCase):
     def setUp(self):
-        self.player_red = Player('red', FirstNonFullColumnStrategy())
-        self.player_blue = Player('blue', EmptiestColumnStrategy())
-        self.match = Match(self.player_red, self.player_blue, GameBoard())
+        self.player_white = Player('W', FirstNonFullColumnStrategy())
```
Comment (basicNew) [inline tests/test_match.py]: player_white -> white_player. Same with black

**Review by basicNew — COMMENTED** [review]
Couple of minor comments

File: tests/test_match.py [hunk 3]
```diff
@@ -9,66 +9,66 @@
...
-                    [None,   'red',  'red',  'red',   None,   None, None],
-                    ['red', 'blue',  'red', 'blue',   None,   None, None],
-                    ['blue', 'red', 'blue',  'red',   None,   None, None],
-                    ['blue', 'red',  'red', 'blue',   None,   None, None],
-                    ['red', 'blue',  'red', 'blue',   None,   None, None],
-                    ['blue', 'red', 'blue',  'red',  'red',  'red', None]]))
-        self.assertFalse(match_red_win_from_row.is_over())
-        match_red_win_from_row.play_next_turn()
-        self.assertTrue(match_red_win_from_row.is_over())
+                    [None, 'B', 'B', 'B', None, None, None],
```
Comment (basicNew) [inline tests/test_match.py]: I thought we were removing all references to `from_matrix`

File: tests/test_gameboard.py [hunk 4]
```diff
@@ -6,311 +6,355 @@
...
-        self.assertEqual(self.board_test_diagonals.winner_exists(), True)
+        test_board = self.builder.build_from_moves([7,6,7,6,7,6,6,7,2,5,5,2,5,5,1,4,4,3])
+        """
+        . . . . . . .
+        . . . . B W B
+        . . . . W B W
+        . B . W W B W
+        W W B B B B W
+        """
+        test_board.put_chip(7, 'W')
```
Comment (basicNew) [inline tests/test_gameboard.py]: Just to make sure the winner didn't exist before, you should add an `assertFalse(test_board.winner_exists())` before the `put_chip`

File: tests/test_gameboard.py [hunk 5]
```diff
@@ -6,311 +6,355 @@
...
+        self.assertEqual(test_board.winner_exists(), True)
```
Comment (basicNew) [inline tests/test_gameboard.py]: Use `assertTrue` instead of `assertEquals`

**Review by basicNew — COMMENTED** [review]
Couple more comments

**Comment by apojomovsky** [comment]
This is ready for another pass @basicNew

File: game/gameboard.py [hunk 6]
```diff
@@ -45,12 +45,8 @@ def __str__(self):
         representation = ''
         for row in self.get_rows():
             for entry in row:
-                if entry is None:
-                    representation = representation + '. '
-                elif entry == 'red':
-                    representation = representation + 'R '
-                elif entry == 'blue':
-                    representation = representation + 'B '
+                if entry is None: entry = "."
```
Comment (basicNew) [inline game/gameboard.py:48]: A bit hacky for my taste, but fair enough :)

File: game/gameboard.py [hunk 7]
```diff
@@ -387,31 +360,3 @@ def _array_contains(self, array_small, array_big):
...
-            indices = [index for index, entry in enumerate(column) if entry == None]
-            if indices:
-                if indices != [number for number in range(len(indices))]:
-                    return False
-        return True
-
-    def _is_valid_cell_value(self, entry):
-        """Checks if a given value is a valid entry"""
-        valid_entries = (None, 'blue', 'red')
-        return entry in valid_entries
```
Comment (basicNew) [inline game/gameboard.py:420]: Isn't it nice to remove all that code? :)

**Review by basicNew — APPROVED** [review]
LGTM, feel free to merge

**Pull request merged**